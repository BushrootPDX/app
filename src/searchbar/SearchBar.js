import React from 'react';

export default function SearchBar(queryField = 'Search Plants Database...', searchFunction) {

    return (
        <form>
            <input type="text" value={queryField}></input>
            <button onSubmit={searchFunction}>Search</button>
        </form>
    );
}