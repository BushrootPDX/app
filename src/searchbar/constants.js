const QUERY_CHANGE = 'QUERY_CHANGE';
const QUERY_SUBMIT = 'QUERY_SUBMIT';
const QUERY_RESET = 'QUERY_RESET';