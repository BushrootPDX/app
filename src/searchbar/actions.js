import * as actions from './constants';

export function queryChange(query='') {
    return {
        type: actions.QUERY_CHANGE,
        payload: query
    };
}

export function queryReset() {
    return {
        type: actions.QUERY_RESET,
        payload: 'Search...'
    };
}