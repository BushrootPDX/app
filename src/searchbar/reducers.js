import * as actions from './constants';

export const queryField = 
    (state = 'Search...', { type, payload }) => {
        switch(type) {
        case actions.QUERY_CHANGE:
            return payload;
        case actions.QUERY_RESET:
            return payload;
        default: 
            return state;
        }
    };
