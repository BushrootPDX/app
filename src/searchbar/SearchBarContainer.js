import { connect } from 'react-redux';
import { queryChange, queryReset } from './actions';
import { queryPlantDb } from '../plantsearch/actions';
import { bindActionCreators } from 'redux';
import SearchBar from './SearchBar';

const mapStateToProps = state => ({
    queryField: state.queryField
});

function mapDispatchToProps (dispatch) {
    return bindActionCreators({ queryChange, queryReset, queryPlantDb }, dispatch);
}

const SearchBarContainer = connect(
    mapStateToProps,
    mapDispatchToProps,
)(SearchBar);

export default SearchBarContainer;