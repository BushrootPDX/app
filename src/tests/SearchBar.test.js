import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { SearchBar } from '../searchbar/SearchBar';

const searchFunction = function(value) {
    console.log('something something', value);
};

describe('SearchBar component', () => {

    it('renders a search bar with a default value', () => {
        const queryField = null;

        const wrapper = shallow(<SearchBar queryField={queryField} searchFunction={searchFunction} />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('renders a search bar with a supplied value', () => {
        const queryField = 'potato';

        const wrapper = shallow(<SearchBar queryField={queryField} searchFunction={searchFunction} />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });


});