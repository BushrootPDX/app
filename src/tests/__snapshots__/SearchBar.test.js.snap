// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders a search bar with a default value 1`] = `
<form>
  <input
    type="text"
    value={
      Object {
        "queryField": null,
        "searchFunction": [Function],
      }
    }
  />
  <button
    onSubmit={Object {}}
  >
    Search
  </button>
</form>
`;

exports[`SearchBar component renders a search bar with a supplied value 1`] = `
<form>
  <input
    type="text"
    value={
      Object {
        "queryField": "potato",
        "searchFunction": [Function],
      }
    }
  />
  <button
    onSubmit={Object {}}
  >
    Search
  </button>
</form>
`;
